apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -345,19 +352,56 @@ export default class CoreLibrary {
     return spriteIds;
   }
 
+  getNumberOfSprites() {
+    return Object.keys(this.nativeSpriteMap).length;
+  }
+
+  getCappedNumSprites(numRequested) {
+    const numSpritesSoFar = this.getNumberOfSprites();
+    // (MAX_NUM_SPRITES + 1) is the actual maximum number of sprites possible
+    // At MAX_NUM_SPRITES, the workspace alert warning is displayed
+    const numNewSpritesPossible = MAX_NUM_SPRITES + 1 - numSpritesSoFar;
+    return Math.min(numRequested, numNewSpritesPossible);
+  }
+
   getLastSpeechBubbleForSpriteId(spriteId) {
     const speechBubbles = this.speechBubbles.filter(
       ({sprite}) => sprite.id === parseInt(spriteId)
     );
     return speechBubbles[speechBubbles.length - 1];
   }
 
+  // When the total number of sprites reaches MAX_NUM_SPRITES,
+  // a workspace alert warning is displayed.
+  // (MAX_NUM_SPRITES + 1) is the actual maximum number of sprites possible
+  // and once this number is reached, this function returns true
+  checkReachSpriteLimit() {
```

Comment:
nit: maybe `hasReachedSpriteLimit` or `reachedSpriteLimit` since this returns a boolean?

Reply:
Makes sense! Will use reachedSpriteLimit.